### PR TITLE
add toggle

### PR DESCRIPTION
**Pull request by kaiwerther**
add toggle based on figma design

### CODE REVIEW

File: src/components/ifxToggle/ifxToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  toggle() {
+    this.checked = !this.checked;
+    this.valueChanged.emit(this.checked);
+
+    console.log(this.checked);
+  }
+
+  render() {
+    return (
+      <div class={`pill-container ${this.checked ? 'checked' : ''}`} onClick={() => this.toggle()}>
```
Comment (tishoyanchev) [inline src/components/ifxToggle/ifxToggle.tsx:23]: Why are the classes prefixed with `pill-`and not `toggle-`?

File: src/components/ifxToggle/ifxToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        <div class={`pill-switch ${this.checked ? 'checked' : ''}`} />
+      </div>
+    );
```
Comment (tishoyanchev) [inline src/components/ifxToggle/ifxToggle.tsx:26]: In the Figma design of the `toggle`, the user should be allowed to add `label` inside two sides of a toggle.

File: src/components/ifxToggle/ifxToggle.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// pill-toggle.tsx
+import { Component, Prop, h, Event, EventEmitter } from '@stencil/core';
+
+@Component({
+  tag: 'ifx-toggle',
+  styleUrl: 'ifxToggle.scss',
+  shadow: true,
+})
+export class ifxToggle {
+  @Prop({ mutable: true }) checked: boolean = false;
```
Comment (tishoyanchev) [inline src/components/ifxToggle/ifxToggle.tsx:10]: `Outline` variant is missing.

File: src/components/ifxToggle/ifxToggle.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+export default {
+  title: 'Components/IfxToggle',
+  args: {
+    checked: true,
+  },
+  argTypes: {
+
+    checked: {
+       control: 'boolean',
```
Comment (tishoyanchev) [inline src/components/ifxToggle/ifxToggle.stories.tsx:11]: Why is the `checked boolean` defined twice?

**Review by tishoyanchev — CHANGES_REQUESTED** [review]

File: src/components/ifxToggle/ifxToggle.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  @Event({ eventName: 'valueChanged' }) valueChanged: EventEmitter<boolean>;
+
+  toggle() {
+    this.checked = !this.checked;
+    this.valueChanged.emit(this.checked);
+
+    console.log(this.checked);
```
[the remark this answers is not in the document]
Comment (kaiwerther) [reply src/components/ifxToggle/ifxToggle.tsx:18]: yes :)

File: src/components/ifxToggle/ifxToggle.tsx (hunk 3 above)
Comment (kaiwerther) [reply src/components/ifxToggle/ifxToggle.tsx:10]: There is no Outline Variant in Figma? (maybe i'm blind)

File: src/components/ifxToggle/ifxToggle.stories.tsx (hunk 4 above)
Comment (kaiwerther) [reply src/components/ifxToggle/ifxToggle.stories.tsx:11]: It isn't - one is the default, one is the definition

File: src/components/ifxToggle/ifxToggle.tsx (hunk 1 above)
Comment (kaiwerther) [reply src/components/ifxToggle/ifxToggle.tsx:23]: copy & paste ;)

File: src/components/ifxToggle/ifxToggle.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+// pill-toggle.tsx
+import { Component, Prop, h, Event, EventEmitter } from '@stencil/core';
+
+@Component({
+  tag: 'ifx-toggle',
+  styleUrl: 'ifxToggle.scss',
+  shadow: true,
+})
+export class ifxToggle {
```
[the remark this answers is not in the document]
Comment (kaiwerther) [reply src/components/ifxToggle/ifxToggle.tsx:9]: discussion in other PR. I'll close this here.

File: src/components/ifxToggle/ifxToggle.tsx (hunk 2 above)
Comment (tishoyanchev) [reply src/components/ifxToggle/ifxToggle.tsx:24]: Switch =/= Toggle
toggle

**Pull request merged**